src/index.ts

```diff
@@ -39,14 +50,19 @@ if (nodeMajor >= 18) {
   uniFetch = globalThis;
 } else if (nodeMajor > 16 || (nodeMajor === 16 && nodeMinor >= 8)) {
   // TODO: Type this as undici once it has the same type signature as the globals
-  uniFetch = require("undici");
+  uniFetch = {
+    fetch: UndiciFetch as unknown as Fetch["fetch"],
+    Headers: UndiciHeaders as unknown as Fetch["Headers"],
+    Request: UndiciRequest as unknown as Fetch["Request"],
+    Response: UndiciResponse as unknown as Fetch["Response"],
+  };
 } else {
   // TODO: Fix this type casting once node-fetch has the same type signature as the globals
   uniFetch = {
-    fetch: require("node-fetch").default as typeof globalThis.fetch,
-    Headers: require("node-fetch").Headers as typeof globalThis.Headers,
-    Request: require("node-fetch").Request as typeof globalThis.Request,
-    Response: require("node-fetch").Response as typeof globalThis.Response,
+    fetch: NodeFetch as unknown as Fetch["fetch"],
+    Headers: NodeHeaders as unknown as Fetch["Headers"],
+    Request: NodeRequest as unknown as Fetch["Request"],
+    Response: NodeResponse as unknown as Fetch["Response"],
```

Comment:
Thinking about it it might just be cleaner to type cast the whole object rather than each entry. Doesn't matter much either way.

Reply:
I'm not sure it would be possible, as `node-fetch` has a `fetch`-like `default` export, plus named exports for the other constructors, so in any case we wouldn't be able to use something like `import NodeFetch from "node-fetch"` because `default` needs to be remapped to `fetch`.